source/postprocess/visualization/melt.cc

```diff
@@ -0,0 +1,223 @@
+*/
+
+
+#include <aspect/postprocess/visualization/melt.h>
+#include <aspect/melt.h>
+#include <aspect/material_model/interface.h>
+
+#include <deal.II/numerics/data_out.h>
+
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+
+      template <int dim>
+      MeltMaterialProperties<dim>::
+      MeltMaterialProperties ()
+        :
+        DataPostprocessor<dim> ()
+      {}
+
+      template <int dim>
+      std::vector<std::string>
+      MeltMaterialProperties<dim>::
+      get_names () const
+      {
+        std::vector<std::string> solution_names;
+
+        for (unsigned int i=0; i<property_names.size(); ++i)
+          if (property_names[i] == "fluid density gradient")
+            for (unsigned int i=0; i<dim; ++i)
+              solution_names.push_back ("fluid_density_gradient");
+          else
+            {
+              solution_names.push_back(property_names[i]);
+              std::replace(solution_names.back().begin(),solution_names.back().end(),' ', '_');
+            }
+
+        return solution_names;
+      }
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      MeltMaterialProperties<dim>::
+      get_data_component_interpretation () const
+      {
+        std::vector<DataComponentInterpretation::DataComponentInterpretation> interpretation;
+        for (unsigned int i=0; i<property_names.size(); ++i)
+          {
+            if (property_names[i] == "fluid density gradient")
+              {
+                for (unsigned int c=0; c<dim; ++c)
+                  interpretation.push_back (DataComponentInterpretation::component_is_part_of_vector);
+              }
+            else
+              interpretation.push_back (DataComponentInterpretation::component_is_scalar);
+          }
+
+        return interpretation;
+      }
+
+      template <int dim>
+      UpdateFlags
+      MeltMaterialProperties<dim>::
+      get_needed_update_flags () const
+      {
+        return update_gradients | update_values  | update_q_points;
+      }
+
+      template <int dim>
+      void
+      MeltMaterialProperties<dim>::
+      compute_derived_quantities_vector (const std::vector<Vector<double> >              &uh,
+                                         const std::vector<std::vector<Tensor<1,dim> > > &/*duh*/,
+                                         const std::vector<std::vector<Tensor<2,dim> > > &/*dduh*/,
+                                         const std::vector<Point<dim> >                  &/*normals*/,
+                                         const std::vector<Point<dim> >                  &evaluation_points,
+                                         std::vector<Vector<double> >                    &computed_quantities) const
+      {
+        AssertThrow(this->include_melt_transport()==true, ExcMessage("include_melt_transport has to be on when using melt transport postprocessors."));
```

Comment:
maybe reference the name of the input parameter, instead of the internal variable, in the error message?

Please break onto separate lines.